server/src/main/java/io/pravega/schemaregistry/rules/jsoncompatibility/ArrayTypeComparator.java

```diff
@@ -0,0 +1,133 @@
+package io.pravega.schemaregistry.rules.jsoncompatibility;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.Iterators;
+import io.pravega.schemaregistry.rules.CompatibilityChecker;
+
+import java.util.Iterator;
+
+import static io.pravega.schemaregistry.rules.jsoncompatibility.BreakingChangesStore.BreakingChanges;
+
+public class ArrayTypeComparator {
+    JsonCompatibilityChecker jsonCompatibilityChecker;
+    public void setJsonTypeComparator() {
+        this.jsonCompatibilityChecker = new JsonCompatibilityChecker();
+    }
+    public BreakingChanges compareArrays(JsonNode toCheck, JsonNode toCheckAgainst) {
+        if (toCheck.isArray() && toCheckAgainst.isArray())
+            return arraySimpleBodyComparision(toCheck, toCheckAgainst);
```

Comment:
do you need to do other checks after arraySimpleBodyComparison? 
if so, returning from here in case where there are no breaking changes as part of this may not be the desired goal. right?

Reply:
In [array](https://json-schema.org/understanding-json-schema/reference/array.html) definitions for json, simple body check will not be used. Had added this for `required` and `enum` arrays but seems like this is redundant now since I have other helper methods for them. I am keeping it for the moment to see if it can be used for tuple validation of arrays.

Reply:
you perhaps misunderstood my point. you are "returning" after doin array simple body comparisons.. i am sure there are other checks that need to be done if this is true. if its false, its ok to return. but if its true, shouldnt you move to the next check and perform that. right?